packages/fetchai/skills/oracle_aggregation/strategy.py

```diff
@@ -0,0 +1,316 @@
+            )
+        }
+        self._radius = kwargs.pop("search_radius", DEFAULT_SEARCH_RADIUS)
+
+        self._oracle_request = DEFAULT_ORACLE_REQUEST
+        self._round = 0
+        self._peers = {}
+        self._observations = {}
+        self._aggregation = None
+
+        self._set_personality_data = kwargs.pop(
+            "personality_data", DEFAULT_PERSONALITY_DATA
+        )
+        enforce(
+            len(self._set_personality_data) == 2
+            and "piece" in self._set_personality_data
+            and "value" in self._set_personality_data,
+            "personality_data must contain keys `key` and `value`",
+        )
+        self._set_classification = kwargs.pop("classification", DEFAULT_CLASSIFICATION)
+        enforce(
+            len(self._set_classification) == 2
+            and "piece" in self._set_classification
+            and "value" in self._set_classification,
+            "classification must contain keys `key` and `value`",
+        )
+        self._set_service_data = kwargs.pop("service_data", DEFAULT_SERVICE_DATA)
+        enforce(
+            len(self._set_service_data) == 2
+            and "key" in self._set_service_data
+            and "value" in self._set_service_data,
+            "service_data must contain keys `key` and `value`",
+        )
+        self._remove_service_data = {"key": self._set_service_data["key"]}
+        self._simple_service_data = {
+            self._set_service_data["key"]: self._set_service_data["value"]
+        }
+
+        super().__init__(**kwargs)
+        self._ledger_id = (
+            ledger_id if ledger_id is not None else self.context.default_ledger_id
+        )
+        if currency_id is None:
+            currency_id = self.context.currency_denominations.get(self._ledger_id, None)
+            enforce(
+                currency_id is not None,
+                f"Currency denomination for ledger_id={self._ledger_id} not specified.",
+            )
+        self._currency_id = currency_id
+        self._is_searching = True
+        self._balance = 0
+
+    @property
+    def ledger_id(self) -> str:
+        """Get the ledger id."""
+        return self._ledger_id
+
+    @property
+    def is_ledger_tx(self) -> bool:
+        """Check whether or not tx are settled on a ledger."""
+        return self._is_ledger_tx
+
+    @property
+    def is_searching(self) -> bool:
+        """Check if the agent is searching."""
+        return self._is_searching
+
+    @is_searching.setter
+    def is_searching(self, is_searching: bool) -> None:
+        """Check if the agent is searching."""
+        enforce(isinstance(is_searching, bool), "Can only set bool on is_searching!")
+        self._is_searching = is_searching
+
+    @property
+    def balance(self) -> int:
+        """Get the balance."""
+        return self._balance
+
+    @balance.setter
+    def balance(self, balance: int) -> None:
+        """Set the balance."""
+        self._balance = balance
```

Comment:
You have a lot of unused properties here as a result of copy & paste :).

Reply:
Indeed, I got rid of a lot but missed a few. And I expect we might eventually want to bring back some of the ledger stuff, but for now it's probably better to keep things as light as possible.